Algorithmia/CLI.py

```diff
@@ -161,75 +161,71 @@ def rm(self, path, client):
+            responseContent = response.content
+            if isinstance(responseContent, six.binary_type):
+                responseContent = responseContent.decode()
 
-                content = json.loads(responseContent)
+            content = json.loads(responseContent)
 
-                if 'files' in content:
-                    f = client.file(path)
-                    for file_info in content['files']:
-                        if(file_info['filename'] == file[-1]):
-                            f.set_attributes(file_info)
+            if 'files' in content:
+                f = client.file(path)
+                for file_info in content['files']:
+                    if file_info['filename'] == file[-1]:
+                        f.set_attributes(file_info)
 
-                if(l):
+            if longlist:
+                listing += f.last_modified.strftime("%Y-%m-%d %H:%M:%S") + '   '
+                listing += str(f.size) + '   '
+                listing += f.path + "\n"
+            else:
+                listing += f.path + "\n"
+        else:
+            # path is a directory
+            if longlist:
+                listingDir = client.dir(path)
+                for f in listingDir.dirs():
+                    listing += f.path + "/\n"
+                for f in listingDir.files():
                     listing += f.last_modified.strftime("%Y-%m-%d %H:%M:%S") + '   '
                     listing += str(f.size) + '   '
-                    listing += 'data://'+f.path + "\n"
-                else:
-                    listing += 'data://'+f.path + "\n"
-            else:
-                #path is a directory
-                #long listing
-                if(l):
-
-                    listingDir = client.dir(path)
-                    for f in listingDir.files():
-                        listing += f.last_modified.strftime("%Y-%m-%d %H:%M:%S") + '   '
-                        listing += str(f.size) + '   '
-                        listing += 'data://'+f.path + "\n"
-                    for f in listingDir.dirs():
-                        listing += 'data://'+f.path + "\n"
+                    listing += f.path + "\n"
 
-                else:
-                    listingDir = client.dir(path)
-                    for f in listingDir.list():
-                        listing += 'data://'+f.path + "\n"
-        else:
-            print("operand must be a path to a remote data source data://")
+            else:
+                listingDir = client.dir(path)
+                for f in listingDir.dirs():
+                    listing += f.path + "/\n"
+                for f in listingDir.files():
+                    listing += f.path + "\n"
 
         return listing
 
     # algo cat <file>
     def cat(self, path, client):
         result = ""
         for f in path:
-            if('data://' in f):
-                if(f[-1] == '*'):
+            if '://' in f and not f.startswith("http"):
```

Comment:
honestly I'm in favor of this structure over the regex commands we were using in the past. This is much more readable and achieves roughly the same goal.